pkg/stis/calstis/cs1/dodqi.c

```diff
@@ -652,6 +653,8 @@ int doppmin, doppmax  i: offsets for Doppler shift
 	    xlength = 2;
 	    ylength = tabrow->length * 2;
 	}
+	else
+	    assert(0);
```

Comment:
Asserts are used not to inform the user but to inform the developer (though probably via the user) when they have miscomputed the boundaries to their logic. E.g. in the cases where the developer is pretty sure that some logic will always be true but adds an assert just to be safe. When the assert is triggered at runtime the process exits gracefully (think of this as more of a nice by-product rather than the main feature) and prints the asserted expression, the culprit file and line number (the main feature).

```
#include <assert.h>

int main(void)
{
  int myLogicIsFaultless = 0;
  int whatEver = 1;
  assert(myLogicIsFaultless == whatEver);
  return 0;
}
```

Produces
```
Assertion failed: (myLogicIsFaultless == whatEver), function main, file asserttest.c, line 7.
Abort trap: 6
```